### PR TITLE
 [IMP] helpdesk_mgmt: Changes in email to send and reply mail of the …

### PR DESCRIPTION
**Pull request by ikapasi-initos**
…ticket

### CODE REVIEW

**Comment by ikapasi-initos** [comment]
@jans23 FYI

File: helpdesk_mgmt/models/res_company.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+from odoo import api, fields, models
+
+
+class ResCompany(models.Model):
+    _inherit = "res.company"
+
+    helpdek_from_email = fields.Char(
+        "From email",
+        help="From Email used when sending helpdesk email from Odoo",
+        default="Nitrokey <reply@nitrokey.com>",
```
Comment (manengel) [inline helpdesk_mgmt/models/res_company.py]: I am not sure if that should be the default. I think rather not.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml [hunk 2]
```diff
@@ -0,0 +1,310 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <template id="view_email_template_helpdesk_ticket">
+    <head>
+        <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+        <title>[Nitrokey] Helpdesk Ticket</title>
```
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:6]: Must possibly be written in a more general way.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+        <t t-set="language" t-value="object.partner_id.lang" />
+
+        <table
+                style="font-family: Verdana,sans-serif; font-size: 11px; width: 620px; margin-left: 15px;"
+            >
+            <tbody>
+                <tr>
+                    <td align="left"><img
+                                alt="Nitrokey"
+                                src="/nitrokey_reports/static/src/img/logo.png"
```
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:25]: Will not suit for everybody.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+                                <span
+                                    >we have received your inquiry and will answer it personally as soon as possible.</span><br
+                                    /><br />
+                                <span><a
+                                            href="https://www.nitrokey.com/documentation/frequently-asked-questions-faq"
+                                        >Frequently asked questions</a> are already answered on our website. For technical matters, we provide a lot of information in our <a
+                                            href="https://docs.nitrokey.com/"
+                                        >documentation</a>. You can also get more help and answers in our <a
+                                            href="https://support.nitrokey.com/"
+                                        >forum</a>.</span><br /><br />
```
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:65]: This should be made optional and customizable / generic.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml [hunk 5]
```diff
@@ -0,0 +1,310 @@
...
+                                <span
+                                    >You can reply to this email to add information to the ticket.</span><br
+                                    /><br />
+                                <span>Best regards</span><br />
+                                <span>Your Nitrokey Team</span><br />
```
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:70]: Same as above.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml [hunk 6]
```diff
@@ -0,0 +1,310 @@
...
+                            /><br /><br />
+                    </td>
+                </tr>
+            </table>
+        </div><br /><br />
+        </body>
+    </template>
+
+    <record id="nitrokey_created_ticket_template" model="mail.template">
+        <field name="name">[Nitrokey] Helpdesk Created Ticket Notification Email</field>
```
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:123]: Needs to be more dynamic in regard of the company name.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml [hunk 7]
```diff
@@ -0,0 +1,310 @@
...
+        <t t-set="language" t-value="object.partner_id.lang" />
+
+        <table
+                    style="font-family: Verdana,sans-serif; font-size: 11px; width: 620px; margin-left: 15px;"
+                >
+            <tbody>
+                <tr>
+                    <td align="left"><img
+                                    alt="Nitrokey"
+                                    src="/nitrokey_reports/static/src/img/logo.png"
```
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:156]: Must be replaced with something more generic.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml [hunk 8]
```diff
@@ -0,0 +1,310 @@
...
+                                            >Frequently asked questions</a> are already answered on our website. For technical matters, we provide a lot of information in our <a
+                                                href="https://docs.nitrokey.com/"
+                                            >documentation</a>. You can also get more help and answers in our <a
+                                                href="https://support.nitrokey.com/"
+                                            >forum</a>.</span><br /><br />
+                                <span
+                                        >You can reply to this email to add information to the ticket.</span><br
+                                        /><br />
+                                <span>Best regards</span><br />
+                                <span>Your Nitrokey Team</span><br />
```
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:201]: Also too specific.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml [hunk 9]
```diff
@@ -0,0 +1,310 @@
...
+            </table>
+        </div><br /><br />
+        </body>
+        </field>
+    </record>
+
+    <template id="view_email_default_template_helpdesk_ticket">
+    <head>
+        <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+        <title>[Nitrokey] Helpdesk Ticket</title>
```
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:259]: Name must be more dynamic.

File: helpdesk_mgmt/models/helpdesk_ticket.py [hunk 10]
```diff
@@ -251,3 +253,81 @@ def _message_get_suggested_recipients(self):
...
+
+    def action_ticket_send(self):
+        """
+        This function opens a window to compose an email
+        """
+        self.ensure_one()
+        ir_model_data = self.env["ir.model.data"]
+        try:
+            template_id = ir_model_data.check_object_reference(
+                "nitrokey_helpdesk_mgmt", "nitrokey_default_ticket_template"
```
Comment (manengel) [inline helpdesk_mgmt/models/helpdesk_ticket.py:299]: Do not know if the name(s) should stay like that.

File: helpdesk_mgmt/models/res_company.py [hunk 11]
```diff
@@ -0,0 +1,29 @@
...
+
+    helpdek_from_email = fields.Char(
+        "From email",
+        help="From Email used when sending helpdesk email from Odoo",
+    )
+    ticket_idle_period = fields.Integer(
+        "Tickets idle for X days",
+        default=30,
+        help="Idle period for Ticket. \
+        After X days ticket status changed to done automatically.",
```
Comment (manengel) [inline helpdesk_mgmt/models/res_company.py:15]: Possibly rephrase it to something like this: After this amount of days the ticket status will be changed to done automatically.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml (hunk 8 above)
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:201]: User must be able to customize that.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml (hunk 9 above)
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:259]: Same as many other comments.

File: helpdesk_mgmt/data/helpdesk_ticket_mail.xml [hunk 12]
```diff
@@ -0,0 +1,310 @@
...
+            name="email_to"
+        >{{ not object.partner_id and object.partner_email or '' }}</field>
+        <field name="subject">Re: {{ object.number }} {{ object.name }}</field>
+        <field name="partner_to">{{ object.partner_id.id }}</field>
+        <field name="auto_delete" eval="False" />
+        <field name="lang">{{ object.partner_id.lang }}</field>
+        <field name="body_html" type="html">
+            <head>
+                <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
+                    <title>[Nitrokey] Helpdesk Ticket</title>
```
Comment (manengel) [inline helpdesk_mgmt/data/helpdesk_ticket_mail.xml:290]: Also not ideal to have that hardcoded.

**Review by manengel — CHANGES_REQUESTED** [review]
These parts need to become more specific. In some cases, we can directly access the res.company, in other cases we might need something different.
